hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/DeleteAllowanceChecks.java

```diff
@@ -0,0 +1,352 @@
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.REPEATED_SERIAL_NUMS_IN_NFT_ALLOWANCES;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SENDER_DOES_NOT_OWN_NFT_SERIAL_NO;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_NOT_ASSOCIATED_TO_ACCOUNT;
+
+/**
+ * Semantic check validation for {@link com.hederahashgraph.api.proto.java.CryptoDeleteAllowance} transaction
+ */
+@Singleton
+public class DeleteAllowanceChecks {
+	protected final TypedTokenStore tokenStore;
+	protected final AccountStore accountStore;
+	protected final Supplier<MerkleMap<EntityNumPair, MerkleUniqueToken>> nftsMap;
+	protected final GlobalDynamicProperties dynamicProperties;
+
+	@Inject
+	public DeleteAllowanceChecks(
+			final Supplier<MerkleMap<EntityNumPair, MerkleUniqueToken>> nftsMap,
+			final TypedTokenStore tokenStore,
+			final AccountStore accountStore,
+			final GlobalDynamicProperties dynamicProperties) {
+		this.tokenStore = tokenStore;
+		this.nftsMap = nftsMap;
+		this.dynamicProperties = dynamicProperties;
+		this.accountStore = accountStore;
+	}
+
+	/**
+	 * Validates all allowances provided in
+	 * {@link com.hederahashgraph.api.proto.java.CryptoDeleteAllowanceTransactionBody}
+	 *
+	 * @param cryptoAllowances
+	 * 		given crypto allowances to remove
+	 * @param tokenAllowances
+	 * 		given fungible token allowances to remove
+	 * @param nftAllowances
+	 * 		given nft serials allowances to remove
+	 * @param payerAccount
+	 * 		payer for the transaction
+	 * @return validation response
+	 */
+	public ResponseCodeEnum deleteAllowancesValidation(
+			final List<CryptoRemoveAllowance> cryptoAllowances,
+			final List<TokenRemoveAllowance> tokenAllowances,
+			final List<NftRemoveAllowance> nftAllowances,
+			final Account payerAccount) {
+		// feature flag for allowances
+		if (!isEnabled()) {
+			return NOT_SUPPORTED;
+		}
+
+		var validity = commonChecks(cryptoAllowances, tokenAllowances, nftAllowances);
+		if (validity != OK) {
+			return validity;
+		}
+
+		validity = validateCryptoDeleteAllowances(cryptoAllowances, payerAccount);
+		if (validity != OK) {
+			return validity;
+		}
+
+		validity = validateTokenDeleteAllowances(tokenAllowances, payerAccount);
+		if (validity != OK) {
+			return validity;
+		}
+
+		validity = validateNftDeleteAllowances(nftAllowances, payerAccount);
+		if (validity != OK) {
+			return validity;
+		}
+
+		return OK;
+	}
+
+	/**
+	 * Validates all the {@link CryptoRemoveAllowance}s in the
+	 * {@link com.hederahashgraph.api.proto.java.CryptoDeleteAllowance}
+	 * transaction
+	 *
+	 * @param cryptoAllowances
+	 * 		crypto remove allowances list
+	 * @param payerAccount
+	 * 		payer for the transaction
+	 * @return validation response
+	 */
+	ResponseCodeEnum validateCryptoDeleteAllowances(final List<CryptoRemoveAllowance> cryptoAllowances,
+			final Account payerAccount) {
+		final var distinctOwners = cryptoAllowances.stream()
```

Comment:
Would discourage all use of `Stream`s in `handleTransaction()`...

Reply:
Removed